docs/for-developers/technical-overview.md

```diff
@@ -3,71 +3,200 @@ id: technical-overview
+`DripsHub` and the related smart contracts allow users to create and manage real-time streams of funds (Drips) as well as splitting of any funds sent to them (Splits). One can start, alter or end the process of sending their funds at any time with immediate effect. The flow of funds is maintained automatically by the protocol and is continuous over time.
 
-However, for ease of use, a copy of the Drips webapp is also available at <a href="https://drips.radicle.xyz">drips.radicle.xyz</a>
+### Design Principles
 
-The Drips webapp interacts directly with Ethereum (via Infura), as well as IPFS (via Piñata pinning servers) and
-The Graph nodes to query graphQL.
+Anyone who has worked with Ethereum and other EVM-based blockchains knows that there are significant limitations with regards to gas and transaction costs. Because of this, the Drips protocol was carefully designed to be as scalable and gas-efficient as possible. In particular, we wanted Drips to be able to support use cases like sponsorship or streaming memberships, where a single user (or org) may be receiving funds from hundreds or thousands of senders.
 
-## The Radicle/Drips GraphQL Subgraph
+With these requirements in mind, let's consider the case of a receiver collecting funds that have been Dripped to them. The obvious technical design for a streaming system like Drips would be to store a list of all Drips being sent to each receiver and then simply iterate the list whenever collect() is called. However, storing and iterating a list with hundreds or thousands of entries quickly becomes cost-prohibitive from a gas perspective on Etheruem and might even exceed the total gas available in a block in the worst case.
 
-<i>Code for the Radicle/Drips subgraph can be found in the <a href="https://github.com/radicle-dev/radicle-subgraph">Radicle-Subgraph</a> repository.</i>
+Instead, to increase efficiency, `DripsHub` works internally with the concept of `cycles` and all funds being sent to a given recipient for a given cycle are aggregated and stored together as a pooled amount in the DripsHub smart contracts. In fact, for greater efficiency, it is not even the pooled stream amounts themselves that are stored, but rather the "deltas", or changes in amount streamed, from one cycle to the next.
 
-Drips utilizes a graphQL "subgraph" that contains indexed Ethereum data for efficient qurying based on <a href="https://thegraph.com/">The Graph's</a> prootcol. For simplicity of curation/indexing, the drips subgraph is part of the same subgraph that is used by Radicle orgs.
+### Drips Cycles
 
-For Drips, the subgraph is used solely by the Drips Webapp to allow for efficient calcualtion of historical values, summary statistics and for queries that scope over multiple Communities or Users at oncee.
+A cycle has a fixed time interval so that every block is assigned to a cycle based on its timestamp. Cycles are numbered starting with zero.
```

Comment:
I think that this description may not be enough if you've never heard of drips cycles. IMO the missing nuances are:
- cycles have a fixed, immutable duration and they don't overlap
- cycles are attached to timestamps and are user-agnostic, every user sees the same cycles
- cycles are **1-based**, cycle `0` is an internal `null` placeholder. It probably doesn't matter when understanding the protocol anyway.

Maybe it'd be the easiest to present the actual cycle calculation formula `timestamp / cycleSecs + 1`?

Reply:
I took a crack at re-writing this.